packages/components/src/components/tooltip/tooltip.js

```diff
@@ -151,71 +177,183 @@ class Tooltip extends mixin(
     this.tooltip.changeState(
       state,
       Object.assign(detail, { delegatorNode: this.element }),
-      callback
+      () => {
+        // @questions
+        // -- What is the expected behavior if the user clicks on the trigger when the tooltip is already open
+        // -- What's the focus state look like when there is only rich text within a tooltip (no buttons/links/etc)
+        // -- There is a "flash" where the next element (another trigger in the demo) is focused before focus is
+        //      moved to the tooltip. I'm assuming this is due to the debounce. Now that it's triggered on "click"
+        //      for keyboard users do we still want the debounce?
+        this.isHidden = state !== 'shown';
+        this.tooltip.element.setAttribute(
+          'aria-hidden',
+          this.isHidden.toString()
+        );
+        this.element.setAttribute('aria-expanded', (!this.isHidden).toString());
+
+        if (this._handleFocusinListener) {
+          this._handleFocusinListener = this.unmanage(
+            this._handleFocusinListener
+          ).release();
+        }
+
+        if (this._handleClickListener) {
+          this._handleClickListener = this.unmanage(
+            this._handleClickListener
+          ).release();
+        }
+
+        if (this._handleKeydownListener) {
+          this._handleKeydownListener = this.unmanage(
+            this._handleKeydownListener
+          ).release();
+        }
+
+        if (state === 'shown') {
+          this._hookCloseActions(this.element);
+          const focusableNode = this.tooltip.element.querySelector(
+            settings.selectorTabbable
+          );
+          if (focusableNode) {
+            focusableNode.focus();
+          } else {
+            this.tooltip.element.setAttribute('tabindex', '0');
```

Comment:
Did you want to set `-1` to `tabindex` (in `.hbs`) given you are putting focus manually?

Reply:
I made this update but can you explain to me why?  By having the HTML output the `-1` it's kind of deceiving to developers IMO because it's not obvious that that the JavaScript changes it as needed and it seems to be a required attribute (which it's not).

Reply:
@elizabethsjudd Thank you for making the update. Not sure where the question came from, but let me try answering your question... `tabindex="-1"` means that keyboard focus doesn't automatically happen with tab keys, but `.focus()` call works. That said, you don't have to change `tabindex` in order to ensure `.focus()` works, as long as the element has `tabindex="-1"`. (In other words, this line can be removed) Does this make sense?

Reply:
I understand what `tabindex="-1"` and the lack of `tabindex` do.  The use of `tabindex` will only be needed on tooltips that don't have interactive elements and only contain rich text.  That left me with with a couple of different options:
1.  Add documentation and require Carbon users to manually add `tabindex="0"` to tooltips that meet the criteria otherwise disregard the `tabindex` attribute
2. Make it easier for Carbon users but detecting if the tooltip required the attribute with  JS and adding the attribute as needed.

There shouldn't ever be the need to have `tabindex="-1"` on any tooltip as I ONLY focus the tooltip in the rich text only use case where one of the above options should have been considered.

Reply:
Thank you for clarifying @elizabethsjudd - My concern here is specifically on using JavaScript code to set `tabindex="0"`. Instead, I'd like to see the right `tabindex` added in `.hbs` level. Does this make sense?

Reply:
I updated the HBS file and added a warning for developers instead of directly setting the attribute